sympy/core/relational.py

```diff
@@ -172,25 +172,31 @@ def equals(self, other, failing_expression=False):
 
     def _eval_simplify(self, ratio, measure):
         r = self
-        r = r.func(r.lhs.simplify(ratio=ratio, measure=measure),
-                   r.rhs.simplify(ratio=ratio, measure=measure))
-        if r not in (S.true, S.false):
-            if isinstance(r.lhs, Expr) and isinstance(r.rhs, Expr):
-                dif = r.lhs - r.rhs
-                # We want a Number to compare with zero and be sure to get a
-                # True/False answer.  Check if we can deduce that dif is
-                # definitively zero or non-zero.  If non-zero, replace with an
-                # approximation.  If .equals(0) gives None, cannot be deduced.
-                if not dif.has(Symbol):
-                    know = dif.equals(0)
-                    if know == True:
-                        dif = S.Zero
-                    elif know == False:
-                        dif = dif.evalf()
-                # Can definitively compare a Number to zero, if appropriate.
-                if dif.is_Number and (dif.is_real or r.func in (Eq, Ne)):
-                    # Always T/F (we never return an expression w/ the evalf)
-                    r = r.func._eval_relation(dif, S.Zero)
+        r = r.func(*[i.simplify(ratio=ratio, measure=measure)
+            for i in r.args])
+        if r.is_Relational:
+            dif = r.lhs - r.rhs
+            v = None
+            # replace dif with a valid Number that will
+            # allow a definitive comparison with 0
+            if dif.is_comparable:
+                v = dif.n(2)
+            else:
+                if r.func in (Eq, Ne):  # XXX why does isinstance(Eq(x,0).func, Eq) fail?
+                    a, b = [i.n(2) for i in dif.as_real_imag()]
```

Comment:
@asmeurer - you commented on having to know that `d` is real in `a + d*I` in order to say that `Eq(a + d*I, 0)` is false when `a != 0`. I think that is handled by the code I have here: `a` and `b` (referred to as `d` in the preceding) are real and either the result for `a` is 1) written in terms of `re`/`im` if `dif` couldn't be broken into real and imaginary parts and the `is_comparable` should fail or 2) it is comparable and, if not zero, allows the `Eq` to return False.  I'll see if I can come up with a test for case 1.